### DISCUSSION TITLE
Idea to manage flag similarly to enum

### DISCUSSION THREAD

**Discussion by Hor-us** [vlang/v#11718] [Ideas]
We can a reserved word as flag
```
flag TextAlign {
left
right
center
top
middle
bottom
}
```

this is managed by the compiler as

```
enum TextAlign {
unset = 0
left = 1 << 1
right = 1 << 2
center = 1 << 3
...
}
```
instead 0,1,2,3,4.

The variable defined as 
```
align TextAlign
```
is managed by the compiler as: typedef TextAlign byte|u16|y32... according to the number of flags.
And we can check the flags with the match statement as
```
match alin {
.left {}
.right|.middle {}
else {}
}
```
Which is transformed by the compiler as: 
```
if (align & TextAlign.left { }
```
instead the usual enum (if align == .left)

The cost is very low, similar to the enum management but make very simple the management of flags/states. Traslating all the ugly work to the compiler.

Is it possible?

**Comment by danieldaeschle** [comment]
We could have a attribute [flags] like C# @medvednikov

**Comment by dumblob** [comment]
Related (partially overlaps) https://github.com/vlang/v/issues/1834 .

**Comment by larpon** [comment]
It could be a tag as well?
```
[flag]
TextAlign {
    left
    right
    center
    top
    middle
    bottom
}
```
